packages/webapp/src/components/RepeatCropPlan/index.jsx

```diff
@@ -119,16 +126,17 @@ export default function PureRepeatCropPlan({
     }
 
     const getAndSetMonthlyOptions = async () => {
-      // Store which pattern is currently selected
-      const currentSelection = monthlyOptions?.findIndex(
+      // Utility function uses rrule to generate natural language strings
+      const options = await calculateMonthlyOptions(planStartDate, repeatFrequency);
+
+      // If already on this screen, store which pattern is currently selected.
+      // When returning from next screen, store the selected option
+      const currentSelection = [monthlyOptions.length ? monthlyOptions : options]?.findIndex(
```

Comment:
Nice! This is way more succinct than what I wrote!

I think it has to be parentheses and not brackets though or it doesn't work:

`(monthlyOptions.length ? monthlyOptions : options)?.findIndex(...`

Reply:
Oh no! What was I doing 🤦‍♀️🤦‍♀️🤦‍♀️ Thank you so much for catching it!!!